src/client/components/builder/LogicTab.tsx

```diff
@@ -65,10 +65,12 @@ const generateDefaultDateOp = (id: number): checker.Operation => ({
 })
 
 export const LogicTab: FC = () => {
-  const [activeIndex, setActiveIndex] = useState<number>(0)
+  const { dispatch, config } = useCheckerContext()
+  const [activeIndex, setActiveIndex] = useState<number>(() =>
+    config.operations.length > 0 ? 0 : -1
```

Comment:
There's a bug with this implementation where if we created 2 Logic tab components (where O1 is at index 0 and O2 is at index 1) and then proceeded to delete the first component (O1 at index 0), the `move down` button for the last remaining component is still active. And if we clicked on that `move down` button, an error will be thrown.

Reply:
This actually happens in all 3 tabs^

Reply:
https://github.com/opengovsg/checkfirst/blob/9d0270b0b068767d27d22b4c61ea3120a27dc954/src/client/components/builder/BuilderField.tsx#L199-L213

I think we need to do a check in the `handleDelete` function in `BuilderField.tsx`, where we check both the length of the array corresponding to `configArrName` and current `index`. The current issue is that we don't pass the actual array into the `handleDelete` function, otherwise we could do:

#### Could be represented using a chain of ternary operators
```
index > 0 ? setActiveIndex(index - 1) : (configArr.length === 1 ? setActiveIndex(-1) : setActiveIndex(0))
```

#### Or nested if/else statements
```
    if (index === 0) {
      if (configArr.length === 1) {
        setActiveIndex(-1)
      } else {
        setActiveIndex(0)
      }
    } else {
      setActiveIndex(index - 1)
    }
```

Reply:
Thanks for catching this! Will fix.